### PR TITLE
Unify remaining catalog not found error messages and types

### PR DESCRIPTION
**Pull request by charlesjmorgan**
## Description
I found a few more stragglers that I had missed previously as a part of #20470

## Additional context and related issues
#20470

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (mosabua) [inline core/trino-main/src/main/java/io/trino/Session.java:340]: Sorry .. but what is that message supposed to mean? Is this about a catalog session property in a specific catalog? .. and what is not found .. the catalog or the catalog session property? 

I realize the old message makes no sense btw.. but the new one makes even less sense.

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (mosabua) [inline core/trino-main/src/main/java/io/trino/Session.java:351]: Can we reword? What is this supposed to mean "Catalog for role?"

**Review by mosabua — CHANGES_REQUESTED** [review]

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (charlesjmorgan) [reply core/trino-main/src/main/java/io/trino/Session.java:340]: I think the catalog is not found, but it's in the context of parsing session properties for a catalog. So the user has session properties set for a catalog, and the catalog was not found. I'm not entirely sure what a better message here would be. I think that `Catalog 'x' not found` would be clearer, but then you lose the context that it was not found while validating the session properties.

Comment (charlesjmorgan) [reply core/trino-main/src/main/java/io/trino/Session.java:340]: Maybe `Catalog 'x' not found while validating session properties`? Or is that too verbose?

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (charlesjmorgan) [reply core/trino-main/src/main/java/io/trino/Session.java:351]: I think it's the roles that have access to the catalog? Also as a side note, the naming of `catalogRoles`/`connectorRoles` is not consistent in the `io.trino.spi.security.Identity` class

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (mosabua) [reply core/trino-main/src/main/java/io/trino/Session.java:340]: Does this come up when the catalog is not found or when the catalog session property is invalid for that catalog? Or both? Depending on scenario we might need different message.

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (mosabua) [reply core/trino-main/src/main/java/io/trino/Session.java:351]: The whole catalog vs connector thing is inconsistent.

For the error message we have to figure out the details. Please let me know here and I can try to come up with a message text.

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (charlesjmorgan) [reply core/trino-main/src/main/java/io/trino/Session.java:340]: When the catalog is not found

Comment (mosabua) [reply core/trino-main/src/main/java/io/trino/Session.java:340]: Lets maybe go with 

Catalog x for session property y not found

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (mosabua) [reply core/trino-main/src/main/java/io/trino/Session.java:351]: I  would just go with

Catalog not found


since we dont say what role anyway and depending on the access control there might be other things at play

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (charlesjmorgan) [reply core/trino-main/src/main/java/io/trino/Session.java:340]: There isn't one particular session property, it groups them by catalog, so it's ALL of the session properties defined for a catalog. Maybe "Catalog 'x' in session properties not found"?

File: core/trino-main/src/main/java/io/trino/Session.java
Comment (mosabua) [inline core/trino-main/src/main/java/io/trino/Session.java:340]:
```suggestion
                    .orElseThrow(() -> new TrinoException(CATALOG_NOT_FOUND, "Catalog '%s' not found".formatted(catalogName)));
```

**Review by mosabua — APPROVED** [review]

**Comment by mosabua** [comment]
Looks good now @charlesjmorgan .. thanks. I will just wait for CI to finish.

**Comment by charlesjmorgan** [comment]
Thanks for the help! @mosabua

**Comment by mosabua** [comment]
Seems like there is at least one failure .. let know what you find @charlesjmorgan

**Comment by charlesjmorgan** [comment]
> Seems like there is at least one failure .. let know what you find @charlesjmorgan

Looks unrelated to me, 503 service unavailable on google sheets connector tests. Would probably pass if rerun

**Comment by charlesjmorgan** [comment]
Bump @mosabua

**Review by mosabua — APPROVED** [review]
Looks good now.

**Comment by mosabua** [comment]
@ebyhr @martint .. this looks good to go to me but I am not sure if its worth having 5 separate commits. One would be just fine with me. Thoughts?

**Comment by charlesjmorgan** [comment]
I was going based on this comment https://github.com/trinodb/trino/pull/20470#issuecomment-1911314176. I'll happily do whatever

**Comment by mosabua** [comment]
Lets roll with the current setup.

**Pull request merged**
